homeassistant/components/isy994/select.py

```diff
@@ -0,0 +1,125 @@
+"""Support for ISY select entities."""
+from __future__ import annotations
+
+from typing import cast
+
+from pyisy import ISY
+from pyisy.constants import (
+    COMMAND_FRIENDLY_NAME,
+    INSTEON_RAMP_RATES,
+    ISY_VALUE_UNKNOWN,
+    PROP_RAMP_RATE,
+    UOM_TO_STATES,
+)
+from pyisy.helpers import NodeProperty
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import _LOGGER, DOMAIN, UOM_INDEX
+from .entity import ISYAuxControlEntity
+
+
+def time_string(i: int) -> str:
+    """Return a formatted ramp rate time string."""
+    if i >= 60:
+        return f"{(float(i)/60):.1f} {UnitOfTime.MINUTES}"
+    return f"{i} {UnitOfTime.SECONDS}"
+
+
+RAMP_RATE_OPTIONS = [time_string(rate) for rate in INSTEON_RAMP_RATES.values()]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up ISY/IoX select entities from config entry."""
+    isy_data = hass.data[DOMAIN][config_entry.entry_id]
+    isy: ISY = isy_data.root
+    device_info = isy_data.devices
+    entities: list[ISYAuxControlIndexSelectEntity | ISYRampRateSelectEntity] = []
+
+    for node, control in isy_data.aux_properties[Platform.SELECT]:
+        name = COMMAND_FRIENDLY_NAME.get(control, control).replace("_", " ").title()
+        if node.address != node.primary_node:
+            name = f"{node.name} {name}"
+
+        node_prop: NodeProperty = node.aux_properties[control]
+
+        options = []
+        if node_prop.uom == UOM_INDEX:
+            if options_dict := UOM_TO_STATES.get(node_prop.uom):
+                options = options_dict.values()
+
+        description = SelectEntityDescription(
+            key=f"{node.address}_{control}",
+            name=name,
+            entity_category=EntityCategory.CONFIG,
+            options=options,
+        )
+        entity_detail = {
+            "node": node,
+            "control": control,
+            "unique_id": f"{isy.uuid}_{node.address}_{control}",
+            "description": description,
+            "device_info": device_info.get(node.primary_node),
+        }
+
+        if control == PROP_RAMP_RATE:
+            description.options = RAMP_RATE_OPTIONS
```

Comment:
It would be better to construct this above since its possible the `SelectEntityDescription` could become frozen in the future